platform/contracts/admin/src/endpoints.rs

```diff
@@ -60,7 +60,15 @@ pub fn migrate(
     }: MigrateMsg,
 ) -> ContractResult<CwResponse> {
     versioning::update_software(deps.storage, CONTRACT_VERSION, Into::into).and_then(|label| {
-        if label == release {
+        ensure_eq!(
+            label,
+            release,
+            ContractError::WrongRelease {
+                reported: label.to_string(),
+                expected: release,
+            }
+        );
+
```

Comment:
please reuse in a real manner, define a `fn check_release(label: ..., expected: ...) -> ContractResult<CwResponse>`